mergify_engine/clients/github.py

```diff
@@ -387,6 +387,13 @@ async def delete(  # type: ignore[override]
             url, **self._prepare_request_kwargs(api_version, oauth_token, **kwargs)
         )
 
+    async def graphql_post(self, query: str) -> typing.Any:
+        response = (await self.post("/graphql", json={"query": query})).json()
+        data = response.get("data")
+        if not data:
+            raise ConnectionError(response)
```

Comment:
ConnectionError is too generic. Create a new exception class like `GraphQLError`.